### PR TITLE
added available options

### PR DESCRIPTION
**Pull request by tusharxoxoxo**
image

### CODE REVIEW

**Comment by tusharxoxoxo** [comment]
fixing #26

File: src/contextgpt_structs.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ impl FromStr for RequestTypeOptions {
             "author" => Ok(RequestTypeOptions::Author),
             "file" => Ok(RequestTypeOptions::File),
             _ => Err(format!(
-                "Could not parse the request type: {}",
+                "Could not parse the request type: {}. Available options are: author, file",
```
Comment (krshrimali) [inline src/contextgpt_structs.rs:18]: While this works right now, the problem is - that if we end up adding one more option, we'll always have to update this message.

Could you think of something where we could have it dynamically generated from the `RequestTypeOptions` enum?

**Review by krshrimali — COMMENTED** [review]
Thank you @tusharxoxoxo for the PR! :) Always great to see someone attempting to fix the TODOs in this project. I've left a comment and looking forward to hearing what you think about it, please let me know if you have any questions.

File: src/contextgpt_structs.rs (hunk 1 above)
Comment (tusharxoxoxo) [reply src/contextgpt_structs.rs]: can you help me with this, I have created a vector and I have added all the RequestTypeOptions to that vector and I am displaying it from there only,

due to which LSP is showing this warning
image

Comment (tusharxoxoxo) [reply src/contextgpt_structs.rs]: I am unable to come up with a solution in which we could dynamically show our enum, can you guide me a bit, the solution I have committed is that we have to manually add all the enums into our vector and then print it out

Comment (krshrimali) [reply src/contextgpt_structs.rs]: Lemme try doing this live and share the solution with you, happy for you to just pick it up from there and try committing it :)

Thanks for making efforts with this, please don't close the PR - let's just modify a few things later and get this in. I appreciate you taking time to contribute!

**Comment by krshrimali** [comment]
Hi @tusharxoxoxo - Lemme know if you'd like to fix anything here, otherwise I think we can get this merged 🚀

**Comment by tusharxoxoxo** [comment]
looks good! Let's go ahead and get it merged. 🚀

**Pull request merged**